docs/source/getting-started/first-feature.rst

```diff
@@ -0,0 +1,143 @@
+.. Copyright (c) 2020 The DAML Authors. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Your First Feature: Messaging Friends
+*************************************
+
+Let's dive into implementing a feature for our social network app.
+From that we'll get a better idea of how to build DAML applications using our template.
+
+Right now our app allows us to add and remove friends, but we can't communicate with them!
+Let's fix that by adding a private messaging feature.
+We will allow a user to send messages to a number of friends at once, and see all the messages that have been sent to them.
+Of course we must make sure that no one can see messages that were not sent to them.
+We will see that DAML lets us implement this in a direct and intuitive way.
+
+There are two parts to building the messaging feature: the DAML code and the UI.
+Let's start with adding to the DAML code, on which we will base our UI changes.
+
+DAML Changes
+============
+
+The DAML code defines the *workflow* of the application.
+This means: what interactions between users (or *parties*) are permitted by the system?
+In the context of our feature, the question is: when is a user allowed to message another user?
+
+The approach we'll take is: a user Bob can message another user Alice if Alice has added Bob as a friend.
+Remember that friendships are single-directional in our app.
+So Alice adding Bob as a friend means that she gives permission (or *authority*) for Bob to send her a message.
+
+In DAML this workflow is represented as a new choice on the ``User`` contract.
+
+.. literalinclude:: code/daml/User.daml
+  :language: daml
+  :start-after: -- SENDMESSAGE_BEGIN
+  :end-before: -- SENDMESSAGE_END
+
+Let's break this down.
+The choice is ``nonconsuming`` because sending a message should not affect the existence of the ``User`` contract.
+By convention, the choice returns the ``ContractId`` of the resulting ``Message`` contract (which we'll show next).
+Next, the parameters to the choice are the sender (the party wishing to talk to the signatory of this ``User`` contract) and the message text.
```

Comment:
```suggestion
The parameters to the choice are the sender (the party wishing to talk to the signatory of this ``User`` contract) and the message text.
```
Otherwise, we'll have two occurrences of the word "next" next to each other.